distributed/worker_state_machine.py

```diff
@@ -3157,13 +3162,24 @@ def validate_state(self) -> None:
                 assert ts.state == "fetch"
                 assert worker in ts.who_has
 
+        # FIXME https://github.com/dask/distributed/issues/6689
+        # for ts in self.executing:
+        #     assert ts.state == "executing" or (
+        #         ts.state in ("cancelled", "resumed") and ts._previous == "executing"
+        #     ), (ts, ts._previous)
+        # for ts in self.long_running:
+        #     assert ts.state == "long-running" or (
+        #         ts.state in ("cancelled", "resumed") and ts._previous == "long-running"
+        #     ), (ts, ts._previous)
```

Comment:
This is the same as #6689 : there are tasks with "error" state in self.executing.

I found `forgotten` tasks in self.executing. I'm persuaded that the cancelled/resumed state is involved.